gtda/diagrams/tests/test_features_representations.py

```diff
@@ -9,38 +9,77 @@
 from numpy.testing import assert_almost_equal
 from sklearn.exceptions import NotFittedError
 
-from gtda.diagrams import PersistenceEntropy, HeatKernel, \
-    PersistenceImage, Silhouette
+from gtda.diagrams import PersistenceEntropy, BettiCurve, \
+    PersistenceLandscape, HeatKernel, PersistenceImage, Silhouette
 
-diagram = np.array([[[0, 1, 0], [2, 3, 0], [4, 6, 1], [2, 6, 1]]])
+X = np.array([[[0., 1., 0.], [2., 3., 0.], [4., 6., 1.], [2., 6., 1.]]])
 
 
-def test_pe_not_fitted():
-    pe = PersistenceEntropy()
+def test_not_fitted():
+    with pytest.raises(NotFittedError):
+        PersistenceEntropy().transform(X)
+
+    with pytest.raises(NotFittedError):
+        BettiCurve().transform(X)
+
+    with pytest.raises(NotFittedError):
+        PersistenceLandscape().transform(X)
+
+    with pytest.raises(NotFittedError):
+        HeatKernel().transform(X)
+
+    with pytest.raises(NotFittedError):
+        PersistenceImage().transform(X)
 
     with pytest.raises(NotFittedError):
-        pe.transform(diagram)
+        Silhouette().transform(X)
+
+
+@pytest.mark.parametrize('hom_dim_ix', [0, 1])
+def test_fit_transform_plot_1(hom_dim_ix):
```

Comment:
Would you mind giving a more descriptive name, or a comment for future selves?
If i understand correctly, we plot a single homology dimension.

Reply:
Some classes have the `homology_dimension_ix` parameter, others `homology_dimensions`. I could not find a short convincing name, but `test_fit_transform_plot_one_hom_dim` `test_fit_transform_plot_single_many_hom_dims` could work with me.

Reply:
👍